src/ResourceManager/Common/Commands.Common.Authentication.ResourceManager/Models/PSAzureEnvironment.cs

```diff
@@ -228,6 +228,11 @@ public bool OnPremise
         public IDictionary<string, string> ExtendedProperties { get; } = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
 
         /// <summary>
+        /// Gets or sets the Azure Batch AD resource ID.
+        /// </summary>
+        public string BatchEndpointResourceId { get; set; }
```

Comment:
No, we cannot add a new value to the environment.  We need to find another way to discover this value